lib/sparkql/expression_resolver.rb

```diff
@@ -0,0 +1,11 @@
+# Base class for handling expression resolution
+class Sparkql::ExpressionResolver
+
+  VALID_RESULTS = [true, false, :drop]
```

Comment:
I know we use it internally as a business rule for searching listings, but outside of that `:drop` isn't obvious.  It might be worth noting in here what `:drop` means, and when to use it.